lib/auth/clusterconfig/clusterconfigv1/service.go

```diff
@@ -245,3 +250,180 @@ func (s Service) ResetAuthPreference(ctx context.Context, req *clusterconfigpb.R
+		if err2 := authzCtx.CheckAccessToKind(types.KindClusterConfig, types.VerbUpdate); err2 != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	// Support reused MFA for bulk tctl create requests.
+	if err := authzCtx.AuthorizeAdminActionAllowReusedMFA(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	tst, err := req.ClusterNetworkConfig.GetTunnelStrategyType()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if tst == types.ProxyPeering && modules.GetModules().BuildType() != modules.BuildEnterprise {
+		return nil, trace.AccessDenied("proxy peering is an enterprise-only feature")
+	}
+
+	updated, err := s.backend.UpdateClusterNetworkingConfig(ctx, req.ClusterNetworkConfig)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	cfgV2, ok := updated.(*types.ClusterNetworkingConfigV2)
+	if !ok {
+		return nil, trace.Wrap(trace.BadParameter("unexpected type %T", updated))
+	}
+
+	return cfgV2, nil
+}
+
+// UpsertClusterNetworkingConfig creates a new networking configuration or overwrites an existing configuration.
+func (s *Service) UpsertClusterNetworkingConfig(ctx context.Context, req *clusterconfigpb.UpsertClusterNetworkingConfigRequest) (*types.ClusterNetworkingConfigV2, error) {
+	authzCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := authzCtx.CheckAccessToKind(types.KindClusterNetworkingConfig, types.VerbCreate, types.VerbUpdate); err != nil {
+		if err2 := authzCtx.CheckAccessToKind(types.KindClusterConfig, types.VerbCreate, types.VerbUpdate); err2 != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	// Support reused MFA for bulk tctl create requests.
+	if err := authzCtx.AuthorizeAdminActionAllowReusedMFA(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	tst, err := req.ClusterNetworkConfig.GetTunnelStrategyType()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if tst == types.ProxyPeering && modules.GetModules().BuildType() != modules.BuildEnterprise {
+		return nil, trace.AccessDenied("proxy peering is an enterprise-only feature")
+	}
+
+	upserted, err := s.backend.UpsertClusterNetworkingConfig(ctx, req.GetClusterNetworkConfig())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	cfgV2, ok := upserted.(*types.ClusterNetworkingConfigV2)
+	if !ok {
+		return nil, trace.Wrap(trace.BadParameter("unexpected type %T", upserted))
+	}
+
+	return cfgV2, nil
+}
+
+// ResetClusterNetworkingConfig restores the networking configuration to the default value.
+func (s *Service) ResetClusterNetworkingConfig(ctx context.Context, _ *clusterconfigpb.ResetClusterNetworkingConfigRequest) (*types.ClusterNetworkingConfigV2, error) {
+	authzCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	cfg, err := s.cache.GetClusterNetworkingConfig(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if cfg.Origin() == types.OriginConfigFile {
+		return nil, trace.BadParameter("config-file configuration cannot be reset")
```

Comment:
I'm assuming the reason we disallow _resetting_ a config-file cnc when we seem to allow updating it is that theres a hierarchy where custom cnc takes precedence over config-file cnc, which takes precedence over default cnc.  Whatever the reason, it would be helpful to have comments to clarify the policy.

Additionally, I think the error message could be a bit friendlier.  Ex:

```suggestion
		return nil, trace.BadParameter("cluster networking configuration has been defined in the auth server's config file and cannot be set back to defaults dynamically.")
```